File: 2022/typescript/src/day01/part1and2.ts
const findElfTotals = (input: string): number[] => {
  const elfBags = input.trim().split("\n\n");
  const elfTotals: number[] = [];

  for (let i = 0; i < elfBags.length; i++) {
    let bag = elfBags[i].split("\n").map((calories) => Number(calories));
    elfTotals.push(sum(bag));
  }

  return elfTotals;
};

export const findHeaviest = (input: string): number => {
  return Math.max(...findElfTotals(input));
};

export const findThreeHeaviest = (input: string): number => {
  const elfTotals = findElfTotals(input).sort((a, b) => b - a);
  return sum(elfTotals.slice(0, 3));
};

const sum = (numbers: number[]): number => {
  return numbers.reduce((sum, nextNumber) => sum + nextNumber);
};

File: 2022/python/day01/part1and2.py
def backpack_amount(backpack_string):
  item_calories = [int(item) for item in backpack_string.split("\n")]
  return sum(item_calories)

def calculate_backpacks(input):
  backpacks = input.rstrip().split("\n\n")
  backpack_amounts = [backpack_amount(pack) for pack in backpacks]
  return backpack_amounts

def find_biggest_backpack(input):
  backpack_amounts = calculate_backpacks(input)
  return max(backpack_amounts)

def find_top_three_backpacks(input):
  backpack_amounts = calculate_backpacks(input)
  backpack_amounts.sort()
  return sum(backpack_amounts[-3:])